### PR TITLE
Issue/3305 geotag extraction by cron

### PR DESCRIPTION
**Pull request by stellanl**
#### Before the PR (what is the issue or what needed to be done)
Some versions of the app did not extract geotags form IMAGE answer files.
#### The solution
Run a cron job once a day to look for IMAGE answers where the geotag has not been extracted, and try to extract it (if the file can be found).
Also improved the datascript from #3180 (that was run once for every instance).

#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [x] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 1]
```diff
@@ -212,4 +231,172 @@ private void purgeOrphanJobQueueRecords() {
...
+     * Intended to be run every day
+     */
+    private void extractImageFileGeotags() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.MONTH, ONE_MONTH_AGO);
+        log.info("Starting scan for image answers, newer than: " + deadline.getTime());
+        QuestionAnswerStoreDao qaDao = new QuestionAnswerStoreDao();
+        Long cursor = null;
+        List<QuestionAnswerStore> dfjqList;
+        int retirees = 0;
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: Is this value actually used?

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 2]
```diff
@@ -212,4 +231,172 @@ private void purgeOrphanJobQueueRecords() {
...
+        int json = 0;
+        int nonjson = 0;
+        Media media;
+
+        do {
+            dfjqList = qaDao.listByTypeAndDate("IMAGE", cursor, deadline.getTime(), null, 1000);
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: What is the cursor for? I checked the method and the second param is surveyId, cursor is the second before last

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java [hunk 3]
```diff
@@ -212,4 +231,172 @@ private void purgeOrphanJobQueueRecords() {
...
+                boolean forceSave = false;
+                String v = item.getValue();
+                log.fine(String.format(" Old IMAGE value '%s'", v));
+                if (v != null && !v.trim().equals("")) {
+                    if (v.startsWith("{")) {
+                        json++;
+                        //Parse it
+                        media = MediaResponse.parse(v);
+                        if (media.getLocation() != null) { //Best case: Already known (could check validity)
+                            //System.out.println("Location in " + q);
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: I would maybe remove this commented out line

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java (hunk 2 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: True. Looked at the javadoc list...

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java (hunk 3 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: Done

File: GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java (hunk 1 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java]: No

**Pull request merged**